frontends/p4/strengthReduction.cpp

```diff
@@ -299,9 +302,13 @@ const IR::Node *DoStrengthReduction::postorder(IR::Mux *expr) {
         return expr->e0;
     else if (isFalse(expr->e1) && isTrue(expr->e2))
         return new IR::LNot(expr->e0);
-    else if (auto lnot = expr->e0->to<IR::LNot>())
-        return new IR::Mux(lnot->expr, expr->e2, expr->e1);
-    else if (!hasSideEffects(expr) && expr->e1->equiv(*expr->e2))
+    else if (const auto *lnot = expr->e0->to<IR::LNot>()) {
+        expr->e0 = lnot->expr;
```

Comment:
I find the original much more readable

Reply:
But it does incur another copy and does not preserve source information and type.